### DISCUSSION TITLE
Array Containing Font Data for Rendering in OpenGL with stb_truetype

### DISCUSSION THREAD

**Discussion by ghost** [nothings/stb#1581] [Q&A]
Hi,

I'm working on a project where I use the stb_truetype library for rendering text in an OpenGL environment. I've encountered a problem related to different methods of loading font data. Here's a detailed description of the situation:

1. Actions Taken:
- I converted the "9x18.ttf" font to a C array using the command `xxd -i 9x18.ttf > font_data.c`.
- Then, I used the `fontData` array, which contains the data of "9x18.ttf" font, in my project.

2. Observed Problem:
- When loading the "9x18.ttf" font directly from the `.ttf` file, the text renders correctly.
- When I use the hard-coded `fontData array` with "9x18.ttf" data, the text does not display in the window.

3. Technical Details:
- I'm using the `stbtt_InitFont` and `stbtt_BakeFontBitmap` functions for font initialization and baking.
Text rendering is done using `stbtt_GetBakedQuad` and standard OpenGL calls.

4. Questions:
- Can different methods of loading "9x18.ttf" font data affect rendering in OpenGL?
- Are there any known issues or limitations in the stb_truetype library that might explain this behavior?

5. Additional Information:
- I have checked that the first bytes of the font data are identical in both cases.
- I ensured that `stbtt_InitFont` returns success in both loading methods.
- I checked the OpenGL settings.

I would appreciate any suggestions or insights that might help me solve this problem.

**Comment by nothings** [comment]
There shouldn't be anything about stb_truetype (or OpenGL) that would cause this.

**Comment by ghost** [comment]


**Reply by nothings** [reply]
i mean, it sounds like formatting, but what is doing the printing and why is it different? these seems very outside the scope of stb_truetype help, though, it sounds like you need help from a general C programming forum or something
